tests/test_run_from_cmdline.py

```diff
@@ -0,0 +1,24 @@
+from unittest.mock import patch
+from dirsync import run
+
+
+def test_run_from_cmd_line_then_create_user_config_file_when_not_exist():
+    with patch.object(run, "os") as os_mock:
+        os_mock.path.isfile.return_value = False
+
+        with patch("builtins.open") as mock_file:
+            with patch.object(run, "sync"), patch.object(run, "ArgParser"):
+                run.from_cmdline()
+
+            assert mock_file.called
+
+
+def test_run_from_cmd_line_error_with_exit_2():
+    with patch.object(run, "os") as os_mock:
+        os_mock.path.isfile.return_value = True
+
+        with patch.object(run, "sync"), patch.object(run, "ArgParser") as arg_parser_mock:
+            arg_parser_mock.side_effect = Exception()
+            with patch.object(run, "sys") as sys_mock:
+                run.from_cmdline()
+                sys_mock.exit.assert_called_with(2)
```

Comment:
Is that test checking that the script exits when the arg parser raises an exception? If yes I'm not sure I understand the point of running it ... but maybe I'm missing something?

Reply:
well, I see this part of the test coverage, without this test the exception context of the try sync is not tested. My aim is always to have 100% coverage on my code, as experience has shown that the bugs usually hide in the few percent not covered.

Reply:
OK, I made that remark believing from_cmdline was already tested (see below) and that there would be quite a lot of more relevant not-covered parts of dirsync that would deserve attention first. But you definitely have a point, and the fact that `from_cmdline` was not tested at all before makes it a lot more relevant.

I indeed realised that the `cmdline` test module where your tests could (and probably should) be located only called `dirsync.run.run`. I think the CmdLineTests.dirsync method should be updated there to call `from_cmdline` instead of `dirsync.run`, and the arguments should be fed into sys.args instead of being passed.

In this module you'll see that the decorator syntax (`@patch(...)`) is preferred to the context manager syntax (`with patch(...)`) for all patches required in the test case. This gets rid of quite a few indents and makes the test a lot more readable.

If you are happy with that, it would be great to:

- modify `tests.cmline.CmdLineTests.dirsync` so that it calls `from_cmdline`
- move your tests in the `tests.cmdline` module and make them inherit from `CmdLineTests`
- use `@patch` on the test method to avoid the context managers in your tests

Sorry for not having had a closer look in my previous review and thanks again for the contributions.